### PR TITLE
Add omni bencher & chain-spec-builder bins to release

### PR DESCRIPTION
**Pull request by EgorPopelyaev**
Closes: https://github.com/paritytech/polkadot-sdk/issues/4354

This PR adds the steps to build and attach `frame-omni-bencher` and `chain-spec-builder` binaries to the release draft

## TODO
- [x] add also chain-spec-builder binary
- [ ] ~~check/investigate Kian's comment: `chain spec builder. Ideally I want it to match the version of the sp-genesis-builder crate`~~ see [comment](https://github.com/paritytech/polkadot-sdk/pull/4518#issuecomment-2134731355)
- [ ] Backport to `polkadot-sdk@1.11` release, so we can use it for next fellows release: https://github.com/polkadot-fellows/runtimes/pull/324
- [ ] Backport to `polkadot-sdk@1.12` release

### CODE REVIEW

File: .github/workflows/release-30_publish_release_draft.yml [hunk 1]
```diff
@@ -23,10 +23,31 @@ jobs:
...
+      - name: Checkout sources
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+      - name: Build frame-omni-bencher binary
+        run: |
+          sudo apt update
+          sudo apt install -y protobuf-compiler
+
+          cargo build --locked --release -p frame-omni-bencher
+          target/release/frame-omni-bencher --version
```
Comment (bkontur) [inline .github/workflows/release-30_publish_release_draft.yml:42]: Let's use production profile

```suggestion
          cargo build --locked --profile=production -p frame-omni-bencher
          target/production/frame-omni-bencher --version
```

File: .github/workflows/release-30_publish_release_draft.yml [hunk 2]
```diff
@@ -23,10 +23,31 @@ jobs:
...
+
+      - name: Upload frame-omni-bencher binary
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        with:
+          name: frame-omni-bencher
+          path: target/release/frame-omni-bencher
```
Comment (bkontur) [inline .github/workflows/release-30_publish_release_draft.yml:48]:
```suggestion
          path: target/production/frame-omni-bencher
```

**Comment by bkontur** [comment]
@kianenigma 
Do we need [here](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/utils/chain-spec-builder/Cargo.toml#L23) this `rlib`?
```
[lib]
crate-type = ["rlib"]
```

**Comment by EgorPopelyaev** [comment]
@kianenigma @bkontur guys, the flow is working now and both bins can be built and attached to the release. 
Regarding the check that the version of the `chain-spec-builder` should match the `sp-genesis-builder`, not sure what to do with that, should be done here as part of the pipeline?

**Comment by kianenigma** [comment]
> @kianenigma Do we need [here](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/utils/chain-spec-builder/Cargo.toml#L23) this `rlib`?
> 
> ```
> [lib]
> crate-type = ["rlib"]
> ```

I don't think so, but I am also not the original author of the crate to know why it was there to begin with. From what I read about the definition of `rlib`, it can be removed. Is it causing issues with the release?

**Review by bkontur — APPROVED** [review]

**Comment by bkontur** [comment]
> > @kianenigma Do we need [here](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/utils/chain-spec-builder/Cargo.toml#L23) this `rlib`?
> > ```
> > [lib]
> > crate-type = ["rlib"]
> > ```
> 
> I don't think so, but I am also not the original author of the crate to know why it was there to begin with. From what I read about the definition of `rlib`, it can be removed. Is it causing issues with the release?

@kianenigma no, there is no problem, it just caught my attention, and we don' use it elsewhere except two binaries:  [`polkadot-cli`](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/cli/Cargo.toml#L18) and [`staging-node-cli`](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/node/cli/Cargo.toml#L36)

and to the same versioning for those two crates, please continue discussions here: https://github.com/paritytech/polkadot-sdk/pull/4518#issuecomment-2134731355

**Review by AndWeHaveAPlan — APPROVED** [review]

**Review by alvicsam — APPROVED** [review]

**Review by pavelsupr — APPROVED** [review]
Everything is OK in this specific commit for me, but the initial release file contained a binary dependency (L. 76) from a single developer without pinning/checks of the integrity of it. IMO it should be localised or replaced.

**Review by kianenigma — APPROVED** [review]

**Pull request merged**